### PR TITLE
Clippy: remove unnecessary borrows

### PR DESCRIPTION
**Pull request by divergentdave**
This fixes some Clippy warnings with a 1.81 toolchain.

### CODE REVIEW

**Review by djc — APPROVED** [review]
Thanks for taking care of this!

**Comment by japaric** [comment]
this can't land until #2427 is merged in #2427 without this PR so I'll cherry pick this into 2430

**Comment by japaric** [comment]
merged in #2427

**Pull request closed without merging**
